Modules/@babylonjs/react-native/windows/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+# cmake 3.15.3+ because of a c++ regression when swift is enable

```

Comment:
I'm confused about the relationship between this CMakeLists.txt and the checked in vcxproj that looks at least partially hand authored.

Reply:
From what i've heard, cmake can't generate a single visual studio solution for multiple platforms/configurations, so we have a hybrid solution:

1. cmake generates BabylonNative dependencies/static libs (for example: BabylonReactNative\Modules\@babylonjs\react-native\submodules\BabylonNative\Build_uwp_x86\ReactNativeBabylon.sln)
2. We manually reference all static libraries for all platforms in the react-native NativeModule project (BabylonReactNative\Modules\@babylonjs\react-native\windows\BabylonReactNative.sln). This allows us to build multiple platform/configurations from the same solution/vcxproj definition.

FWIW, react-native-windows NativeModules take some manual setup/manual edits to vcxproj definitions. We could theoretically do this through cmake, but this requirement in addition to cmake needing to then generate 4+ projects/solutions just to build all the platform flavors we need made it seem like manually setting up a BabylonReactNative.dll project definition was a better developer experience.

Reply:
Ok, so the process right now is to manually run a script or something that runs CMake to generate solutions/projects for each platform/configuration, then build all those, then build the hand crafted solution that references the outputs (built libs) of those generated solutions/projects?

Reply:
yeah thats the general workflow. If you open the BabylonReactNative.sln in Modules\@babylonjs\react-native you can build everything. Playground.sln doesn't currently reference the static lib vcxprojs, so it can only build the BabylonReactNative dll and assumes all libs are already compiled/populated. For packaging, we will ship the source/vcxproj for BabylonReactNative.dll + precompiled libs. I am still tinkering with the vcxproj setup to support this.